Runtime/Services/BacktraceSession.cs

```diff
@@ -0,0 +1,354 @@
+        /// </summary>
+        /// <param name="time">Current game time</param>
+        public void Tick(float time)
+        {
+            _lastUpdateInvoke = time;
+            lock (_object)
+            {
+                // check submission job
+                while (_submissionJobs.Count != 0 && _submissionJobs.First().NextInvokeTime < time)
+                {
+                    var submissionJob = _submissionJobs.Pop();
+                    Send(submissionJob.UniqueEvents, submissionJob.SessionEvents, submissionJob.NumberOfRetries);
+                }
+            }
+            if (_timeIntervalInMs == 0)
+            {
+                return;
+            }
+            lock (_object)
+            {
+
+                var intervalUpdate = (time - _lastUpdateTime) >= _timeIntervalInMs;
+                var reachedEventLimit = _maximumNumberOfEventsInStore == Count() && _maximumNumberOfEventsInStore != 0;
+                if (intervalUpdate == false && reachedEventLimit == false)
+                {
+                    // nothing more to update
+                    return;
+                }
+                _lastUpdateTime = time;
+            }
+            Send();
+        }
+
+        /// <summary>
+        /// Force BacktraceSession to Send data to Backtrace
+        /// </summary>
+        public void Send()
+        {
+            Send(DefaultNumberOfRetries);
+        }
+
+        /// <summary>
+        /// Force BacktraceSession to Send data to Backtrace with retry settings
+        /// </summary>
+        internal void Send(uint numberOfRetries = DefaultNumberOfRetries)
+        {
+            Send(
+               uniqueEvents: UniqueEvents.ToArray(),
+               sessionEvents: SessionEvents.ToArray(),
+               numberOfRetries: numberOfRetries);
+        }
+        private void Send(UniqueEvent[] uniqueEvents, SessionEvent[] sessionEvents, uint numberOfRetries)
+        {
+            if (numberOfRetries == 0)
+            {
+                Debug.LogWarning("Backtrace Session: Cannot send session data to Backtrace due to submission issue.");
+                ClearEvents();
+                return;
+            }
+            if (uniqueEvents.Length + sessionEvents.Length == 0)
+            {
+                return;
+            }
+            var payload = CreateJsonPayload(uniqueEvents, sessionEvents);
+
+            // cleanup existing copy of events
+            ClearEvents();
+
+            // submit data to Backtrace
+            RequestHandler.Post(SubmissionUrl, payload, (long statusCode, bool httpError, string response) =>
+            {
+                if (statusCode == 200)
+                {
+                    OnRequestCompleted();
+                }
+                else if (statusCode == 503)
+                {
+                    _numberOfDroppedRequests++;
+                    if (numberOfRetries - 1 != 0)

```

Comment:
This code is where you want your retry counter to go up instead of down.  If it did, you can just compare directly to `defaultNumberOfRetries` after incrementing, and pass directly to `CalculateNextRetryTime`.  it seems like it would be easy to forget whether or not the number of retries is going up or going down, and almost anything interesting you want to do with retry counters needs them to be going up so I'd rather see us standardize on that.

Reply:
renamed to attemps and now we're increasing variable (not decreasing it like previously)